### PR TITLE
feat(VIST-CPC-1173): Archive completed visit

### PR DESCRIPTION
**Pull request by ioannisp03**
**JIRA:** [Jira link](https://champlainsaintlambert.atlassian.net/browse/CPC-1173)

## Context:

The admin will be able to archive COMPLETED visits if the status is set to completed.
## Does this PR change the .vscode folder in petclinic-frontend?:
No.

## Changes

- Removed previous implementation of "Delete Completed Visit" and replaced with "Archive Completed Visit"
- Added a new 'Status' called 'ARCHIVED'
- Archived Button Pops up when Status is set to "COMPLETED"
- Upon archiving a completed visit, a warning pops up and waits for admin confirmation
- getVisitsForStatus switch case has been fixed by adding 'Break;'
- Made tables collapsible for a cleaner UI

## Before and After UI

BEFORE:
image

------

AFTER:
image
Collapsible table:
image

## Dev notes
The previous implementation was delete completed visits, however it was recently changed to archive completed visits in order to still maintain the visit information (hence some methods have been removed).
## Linked pull requests
[Pull request link](https://github.com/cgerard321/champlain_petclinic/pull/805)

### CODE REVIEW

**Review by AlejandroBrnab — APPROVED** [review]
I like how the UI turned out at the end. It looks more organized when the visits are "archived." Even though before we had them sorted in different tables, they were all shown in the page making it a bit loaded. But now, it looks clean in my opinion. Looks good to me.

File: api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/v2/visit/VisitControllerIntegrationTest.java
Comment (FelixAllard) [inline api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/v2/visit/VisitControllerIntegrationTest.java:1]: I am just curious as to why you deleted those?

File: visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/BusinessLayer/VisitService.java
Comment (FelixAllard) [inline visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/BusinessLayer/VisitService.java:1]: The spacing was needed, good job!

**Review by FelixAllard — APPROVED** [review]
Very good job! I really appreciate the line formatting you added! It makes the code more clear

File: api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/v2/visit/VisitControllerIntegrationTest.java
Comment (ioannisp03) [reply api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/v2/visit/VisitControllerIntegrationTest.java:1]: These were tests that I had created for Delete Completed visits. It is not a feature in the project anymore

**Pull request merged**